src/Stratis.Bitcoin.Tests/Base/ChainRepositoryTest.cs

```diff
@@ -15,6 +15,39 @@ public ChainRepositoryTest() : base(Network.StratisRegTest)
         {
         }
 
+        [Fact]
+        public void FinalizedHeightSavedOnDisk()
+        {
+            string dir = CreateTestDir(this);
+
+            using (var repo = new ChainRepository(dir, new LoggerFactory()))
+            {
+                repo.SaveFinalizedBlockHeightAsync(777).GetAwaiter().GetResult();
+            }
+
+            using (var repo = new ChainRepository(dir, new LoggerFactory()))
+            {
+                repo.LoadFinalizedBlockHeightAsync().GetAwaiter().GetResult();
+                int finalizedHeight = repo.GetFinalizedBlockHeight();
+                Assert.Equal(777, finalizedHeight);
+            }
+        }
+
+        [Fact]
+        public void FinalizedHeightCantBeDecreased()
+        {
+            string dir = CreateTestDir(this);
+
+            using (var repo = new ChainRepository(dir, new LoggerFactory()))
+            {
+                repo.SaveFinalizedBlockHeightAsync(777).GetAwaiter().GetResult();
+                repo.SaveFinalizedBlockHeightAsync(555).GetAwaiter().GetResult();
+
+                int finalizedHeight = repo.GetFinalizedBlockHeight();
+                Assert.Equal(777, finalizedHeight);
```

Comment:
I would also check with new instance that it was safe correctly to the disk just as you do in the previous test.

Reply:
Fixed